### ISSUE TITLE
Can't link against TensorFlowLiteC.framework (iOS) when built on recent TF versions (regression)

### ISSUE THREAD

**Issue by mgalgs** [tensorflow/tensorflow#46442]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OSX 11.1
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): Source
- TensorFlow version (use command below): master
- Python version: 3.8.2
- Bazel version (if compiling from source): 3.7.2 and 3.1.0
- GCC/Compiler version (if compiling from source): clang 12.0.0
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**

It looks like there might be a regression in the build system responsible for generating the `TensorFlowLiteC.framework` iOS library. I've bisected it to 3e9fccb0cb9a, which is a little odd since that's just a bazel version bump... I'm using [am15h/tflite_flutter_plugin](https://github.com/am15h/tflite_flutter_plugin) (which binds directly to the TFLite C API) and can build my own `TensorFlowLiteC.framework` successfully and use it with that plugin up until the aforementioned commit.

From 3e9fccb0cb9a forward (bisection script provided below) I get the following link errors:

<details>
  <summary>Build output</summary>

```
$ flutter build ios
Warning: You are using these overridden dependencies:
! tflite_flutter 0.5.0 from path external/tflite_flutter_plugin
Running "flutter pub get" in RoutespotterApp...                     0.7s
Building com.mgalgs.routespotter for device (ios-release)...
Automatically signing iOS for device deployment using specified development team in Xcode project: PH2D8HJC83
Running pod install...                                              2.3s
Running Xcode build...
Xcode build done.                                            4.3s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    Undefined symbols for architecture arm64:
      "tflite::tensor_utils::MatrixBatchVectorMultiplyAccumulate(signed char const*, int, int, signed char const*, float const*, int, int*, float*,
      tflite::CpuBackendContext*)", referenced from:
          l4840 in TensorFlowLiteC
      "tflite::ops::builtin::Register_FLOOR()", referenced from:
          l4485 in TensorFlowLiteC
      "tflite::ops::custom::Register_MFCC()", referenced from:
          l4485 in TensorFlowLiteC
      "_xnn_reallocate", referenced from:
          l4440 in TensorFlowLiteC
      "_xnn_aligned_allocate", referenced from:
          l4440 in TensorFlowLiteC
      "_xnn_aligned_deallocate", referenced from:
          l4440 in TensorFlowLiteC
      "flatbuffers::EnsureDirExists(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
          l3532 in TensorFlowLiteC
      "flatbuffers::FileExists(char const*)", referenced from:
          l2367 in TensorFlowLiteC
      "flatbuffers::LoadFile(char const*, bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >*)", referenced from:
          l2367 in TensorFlowLiteC
      "flatbuffers::AbsolutePath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
          l2254 in TensorFlowLiteC
      "tflite::tensor_utils::VectorBatchVectorCwiseProductAccumulate(short const*, int, short const*, int, int, int, short*)", referenced from:
          l1730 in TensorFlowLiteC
      "tflite::tensor_utils::VectorScalarMultiply(signed char const*, int, float, float*)", referenced from:
          l1729 in TensorFlowLiteC
      "tflite::tensor_utils::VectorBatchVectorAdd(float const*, int, int, float*)", referenced from:
          l1728 in TensorFlowLiteC
          l1729 in TensorFlowLiteC
      "tflite::tensor_utils::CwiseClipping(short*, int, short)", referenced from:
          l1727 in TensorFlowLiteC
          l1731 in TensorFlowLiteC
      "tflite::tensor_utils::MatrixBatchVectorMultiply(signed char const*, int, signed char const*, int, int, int, int, int, signed char*, signed char)", referenced from:
          l1727 in TensorFlowLiteC
      "tflite::tensor_utils::TwoGateSaturatingAdd(signed char const*, signed char, signed char const*, signed char, int, int, int, int, int, int, short*)", referenced from:
          l1727 in TensorFlowLiteC
      "tflite::tensor_utils::ApplySigmoidFloat(short const*, int, int, short*)", referenced from:
          l1727 in TensorFlowLiteC
      "tflite::tensor_utils::CwiseMul(short const*, short const*, int, int, int, short*)", referenced from:
          l1727 in TensorFlowLiteC
          l1731 in TensorFlowLiteC
      "tflite::tensor_utils::ApplyTanhFloat(short const*, int, int, int, short*)", referenced from:
          l1727 in TensorFlowLiteC
      "tflite::tensor_utils::ApplyTanh(int, short const*, int, int, short*)", referenced from:
          l1726 in TensorFlowLiteC
          l1730 in TensorFlowLiteC
      "tflite::tensor_utils::CwiseMul(short const*, short const*, int, int, int, int, int, signed char*)", referenced from:
          l1726 in TensorFlowLiteC
      "flatbuffers::StripExtension(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
          l814 in TensorFlowLiteC
          l1889 in TensorFlowLiteC
          l3564 in TensorFlowLiteC
      "tflite::tensor_utils::MatrixBatchVectorMultiplyAccumulate(signed char const*, int const*, signed char const*, int, int, int, int, int, int, int*, signed char*,
      tflite::CpuBackendContext*)", referenced from:
          l1726 in TensorFlowLiteC
      "tflite::tensor_utils::CwiseClipping(float*, int, float)", referenced from:
          l1722 in TensorFlowLiteC
          l1724 in TensorFlowLiteC
      "tflite::Subgraph::HasDelegates()", referenced from:
          l1472 in TensorFlowLiteC
      "tflite::Subgraph::IsCancelled()", referenced from:
          l1469 in TensorFlowLiteC
      "tflite::tensor_utils::AsymmetricQuantizeFloats(float const*, int, signed char*, float*, int*)", referenced from:
          l278 in TensorFlowLiteC
          l290 in TensorFlowLiteC
          l414 in TensorFlowLiteC
          l1254 in TensorFlowLiteC
          l1417 in TensorFlowLiteC
          l1724 in TensorFlowLiteC
          l4108 in TensorFlowLiteC
          ...
      "tflite::Subgraph::SetExecutionPlan(std::__1::vector<int, std::__1::allocator<int> > const&)", referenced from:
          l1465 in TensorFlowLiteC
      "tflite::tensor_utils::ApplyLayerNorm(short const*, short const*, int const*, int, int, int, int, int, short*)", referenced from:
          l1726 in TensorFlowLiteC
          l1730 in TensorFlowLiteC
      "_xnn_deallocate", referenced from:
          l4440 in TensorFlowLiteC
      "tflite::Subgraph::SetTensorParametersReadWrite(int, TfLiteType, char const*, unsigned long, int const*, TfLiteQuantization, bool, unsigned long, int const*)",
      referenced from:
          l1462 in TensorFlowLiteC
          l1464 in TensorFlowLiteC
          l3451 in TensorFlowLiteC
      "tflite::Subgraph::SetTensorParametersReadOnly(int, TfLiteType, char const*, unsigned long, int const*, TfLiteQuantization, char const*, unsigned long,
      tflite::Allocation const*, TfLiteSparsity*)", referenced from:
          l1461 in TensorFlowLiteC
          l1463 in TensorFlowLiteC
          l3451 in TensorFlowLiteC
      "tflite::Subgraph::AddTensors(int, int*)", referenced from:
          l1459 in TensorFlowLiteC
          l3454 in TensorFlowLiteC
      "flatbuffers::SaveFile(char const*, char const*, unsigned long, bool)", referenced from:
          l812 in TensorFlowLiteC
          l1892 in TensorFlowLiteC
          l3563 in TensorFlowLiteC
      "tflite::Subgraph::AddNodeWithParameters(std::__1::vector<int, std::__1::allocator<int> > const&, std::__1::vector<int, std::__1::allocator<int> > const&,
      std::__1::vector<int, std::__1::allocator<int> > const&, char const*, unsigned long, void*, TfLiteRegistration const*, int*)", referenced from:
          l1454 in TensorFlowLiteC
          l3443 in TensorFlowLiteC
      "tflite::Subgraph::Invoke()", referenced from:
          l1458 in TensorFlowLiteC
          l3008 in TensorFlowLiteC
          l4742 in TensorFlowLiteC
          l5015 in TensorFlowLiteC
      "tflite::tensor_utils::Sub1Vector(short const*, int, short*)", referenced from:
          l1727 in TensorFlowLiteC
          l1731 in TensorFlowLiteC
      "tflite::Subgraph::ResetVariableTensors()", referenced from:
          l1460 in TensorFlowLiteC
      "tflite::tensor_utils::ApplySigmoid(short const*, int, int, short*)", referenced from:
          l1730 in TensorFlowLiteC
      "flatbuffers::StripPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
          l814 in TensorFlowLiteC
          l1889 in TensorFlowLiteC
          l3564 in TensorFlowLiteC
      "tflite::Subgraph::ReleaseNonPersistentMemory()", referenced from:
          l1457 in TensorFlowLiteC
          l5015 in TensorFlowLiteC
      "tflite::tensor_utils::SparseMatrixBatchVectorMultiplyAccumulate1x4(float const*, int const*, int const*, int, int, float const*, int, float*)", referenced from:
          l1282 in TensorFlowLiteC
          l1286 in TensorFlowLiteC
      "tflite::Subgraph::ResizeInputTensorStrict(int, std::__1::vector<int, std::__1::allocator<int> > const&)", referenced from:
          l1456 in TensorFlowLiteC
      "tflite::tensor_utils::SparseMatrixBatchVectorMultiplyAccumulate(signed char const*, unsigned char const*, int, int, signed char const*, float const*, int, float*)",
      referenced from:
          l1724 in TensorFlowLiteC
          l1729 in TensorFlowLiteC
      "tflite::Subgraph::ResizeInputTensor(int, std::__1::vector<int, std::__1::allocator<int> > const&)", referenced from:
          l1455 in TensorFlowLiteC
          l3007 in TensorFlowLiteC
          l3009 in TensorFlowLiteC
          l4741 in TensorFlowLiteC
      "tflite::tensor_utils::MatrixBatchVectorMultiply(short const*, signed char const*, int, int, int const*, int, int, int, int, signed char*)", referenced from:
          l1727 in TensorFlowLiteC
      "tflite::Subgraph::ReserveNodes(int)", referenced from:
          l1452 in TensorFlowLiteC
          l3443 in TensorFlowLiteC
      "tflite::Subgraph::SetExternalContext(TfLiteExternalContextType, TfLiteExternalContext*)", referenced from:
          l1445 in TensorFlowLiteC
      "tflite::Subgraph::SetCustomAllocationForTensor(int, TfLiteCustomAllocation const&)", referenced from:
          l1446 in TensorFlowLiteC
      "tflite::tensor_utils::MatrixBatchVectorMultiplyAccumulate(float const*, int, int, float const*, int, float*)", referenced from:
          l1252 in TensorFlowLiteC
          l1416 in TensorFlowLiteC
          l1722 in TensorFlowLiteC
          l1728 in TensorFlowLiteC
          l4106 in TensorFlowLiteC
      "tflite::Subgraph::RemoveAllDelegates()", referenced from:
          l1451 in TensorFlowLiteC
          l1470 in TensorFlowLiteC
          l1471 in TensorFlowLiteC
      "tflite::Subgraph::SetVariables(std::__1::vector<int, std::__1::allocator<int> >)", referenced from:
          l1449 in TensorFlowLiteC
          l3454 in TensorFlowLiteC
      "flatbuffers::ConCatPathFileName(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
          l2367 in TensorFlowLiteC
      "tflite::tensor_utils::ReductionSumVector(signed char const*, int*, int, int)", referenced from:
          l1724 in TensorFlowLiteC
          l4108 in TensorFlowLiteC
          l4109 in TensorFlowLiteC
          l4554 in TensorFlowLiteC
          l4864 in TensorFlowLiteC
      "tflite::tensor_utils::CwiseAdd(short const*, short const*, int, int, short*)", referenced from:
          l1727 in TensorFlowLiteC
          l1731 in TensorFlowLiteC
      "tflite::Subgraph::SetOutputs(std::__1::vector<int, std::__1::allocator<int> >)", referenced from:
          l1448 in TensorFlowLiteC
          l3454 in TensorFlowLiteC
      "tflite::Subgraph::Subgraph(tflite::ErrorReporter*, TfLiteExternalContext**, std::__1::vector<std::__1::unique_ptr<tflite::Subgraph,
      std::__1::default_delete<tflite::Subgraph> >, std::__1::allocator<std::__1::unique_ptr<tflite::Subgraph, std::__1::default_delete<tflite::Subgraph> > > >*,
      std::__1::unordered_map<int, std::__1::unique_ptr<tflite::resource::ResourceBase, std::__1::default_delete<tflite::resource::ResourceBase> >, std::__1::hash<int>,
      std::__1::equal_to<int>, std::__1::allocator<std::__1::pair<int const, std::__1::unique_ptr<tflite::resource::ResourceBase,
      std::__1::default_delete<tflite::resource::ResourceBase> > > > >*)", referenced from:
          l1441 in TensorFlowLiteC
      "flatbuffers::PosixPath(char const*)", referenced from:
          l2367 in TensorFlowLiteC
      "tflite::tensor_utils::CwiseClipping(signed char*, int, signed char)", referenced from:
          l1726 in TensorFlowLiteC
          l1727 in TensorFlowLiteC
      "tflite::tensor_utils::Sub1Vector(float const*, int, float*)", referenced from:
          l1722 in TensorFlowLiteC
          l1724 in TensorFlowLiteC
      "tflite::tensor_utils::BatchVectorBatchVectorDotProduct(short const*, short const*, int, int, int*)", referenced from:
          l1418 in TensorFlowLiteC
      "tflite::tensor_utils::MatrixBatchVectorMultiplyAccumulate(signed char const*, int, int, signed char const*, float const*, int, float*, float const*, int const*,
      int*, int*, bool*, tflite::CpuBackendContext*)", referenced from:
          l1254 in TensorFlowLiteC
          l1417 in TensorFlowLiteC
          l1724 in TensorFlowLiteC
          l1729 in TensorFlowLiteC
          l4108 in TensorFlowLiteC
      "tflite::Subgraph::SetCancellationFunction(void*, bool (*)(void*))", referenced from:
          l1468 in TensorFlowLiteC
      "tflite::tensor_utils::ReductionSumVector(int const*, int*, int, int)", referenced from:
          l1418 in TensorFlowLiteC
      "_xnn_allocate", referenced from:
          l4440 in TensorFlowLiteC
      "tflite::tensor_utils::MeanStddevNormalization(float const*, float*, int, int)", referenced from:
          l1728 in TensorFlowLiteC
          l1729 in TensorFlowLiteC
      "tflite::tensor_utils::VectorVectorDotProduct(float const*, float const*, int)", referenced from:
          l1416 in TensorFlowLiteC
          l1417 in TensorFlowLiteC
      "tflite::tensor_utils::IsZeroVector(float const*, int)", referenced from:
          l1254 in TensorFlowLiteC
          l1417 in TensorFlowLiteC
          l1722 in TensorFlowLiteC
          l1724 in TensorFlowLiteC
          l4108 in TensorFlowLiteC
      "tflite::tensor_utils::ApplyLayerNormFloat(short const*, short const*, int, int, int const*, int, int, short*)", referenced from:
          l1727 in TensorFlowLiteC
      "tflite::Subgraph::ModifyGraphWithDelegate(TfLiteDelegate*)", referenced from:
          l1451 in TensorFlowLiteC
          l1470 in TensorFlowLiteC
      "tflite::tensor_utils::MatrixScalarMultiplyAccumulate(signed char const*, int, int, int, int*)", referenced from:
          l764 in TensorFlowLiteC
          l1981 in TensorFlowLiteC
      "tflite::Subgraph::SetInputs(std::__1::vector<int, std::__1::allocator<int> >)", referenced from:
          l1447 in TensorFlowLiteC
          l3454 in TensorFlowLiteC
      "tflite::tensor_utils::ReductionSumVector(float const*, float*, int, int)", referenced from:
          l1416 in TensorFlowLiteC
          l1417 in TensorFlowLiteC
      "tflite::tensor_utils::MatrixBatchVectorMultiplyAccumulate(signed char const*, int const*, signed char const*, int, int, int, int, int, int, int*, short*,
      tflite::CpuBackendContext*)", referenced from:
          l1726 in TensorFlowLiteC
          l1730 in TensorFlowLiteC
      "tflite::tensor_utils::SymmetricQuantizeFloats(float const*, int, signed char*, float*, float*, float*)", referenced from:
          l1254 in TensorFlowLiteC
          l1417 in TensorFlowLiteC
          l1724 in TensorFlowLiteC
          l4108 in TensorFlowLiteC
          l4542 in TensorFlowLiteC
          l4551 in TensorFlowLiteC
          l4837 in TensorFlowLiteC
          ...
      "tflite::Subgraph::AllocateTensors()", referenced from:
          l1450 in TensorFlowLiteC
          l3006 in TensorFlowLiteC
          l3008 in TensorFlowLiteC
          l4741 in TensorFlowLiteC
          l5015 in TensorFlowLiteC
    ld: symbol(s) not found for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.3.99. (in target
    'url_launcher' from project 'Pods')
    warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.3.99. (in target
    'video_player' from project 'Pods')
    warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.3.99. (in target
    'shared_preferences' from project 'Pods')
    warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.3.99. (in target
    'path_provider' from project 'Pods')
    warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.3.99. (in target
    'package_info' from project 'Pods')
    warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.3.99. (in target
    'image_picker' from project 'Pods')
    warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.3.99. (in target
    'flutter_isolate' from project 'Pods')
    warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.3.99. (in target
    'camera' from project 'Pods')
    warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.3.99. (in target
    'Flutter' from project 'Pods')

Encountered error while building for device.
```
</details>

**Describe the expected behavior**

I expect to be able to build a `TensorFlowLiteC.framework` and link against it from a Flutter application.

**Standalone code to reproduce the issue**

These steps must be run on an OSX machine.

If needed, install bazel versions 3.1.0 *and* 3.7.2:

```
(cd "~/.bazel/bin" && curl -fLO https://releases.bazel.build/3.1.0/release/bazel-3.1.0-darwin-x86_64 && chmod +x bazel-3.1.0-darwin-x86_64)
(cd "~/.bazel/bin" && curl -fLO https://releases.bazel.build/3.7.2/release/bazel-3.7.2-darwin-x86_64 && chmod +x bazel-3.7.2-darwin-x86_64)
```

Download [this gist](https://gist.github.com/mgalgs/3a79a6aa4b9ca237e0bea047d2326550) to a `$WORKSPACE` of your choosing, then:

```
cd $WORKSPACE
git clone --recurse-submodules https://github.com/mgalgs/object_detection_flutter.git
cd <tensorflow_dir>
./configure  # answer yes when it asks about iOS support

# Checkout the breaker and you should see a build failure in our test app.
git checkout 3e9fccb0cb9a5f55a4e67e3011c20ff31c3cce67
APPDIR=$WORKSPACE/object_detection_flutter $WORKSPACE/tflite_ios_bisect.sh

# Now checkout the parent of the breaking commit and build again. Should succeed.
git checkout 3e9fccb0cb9a5f55a4e67e3011c20ff31c3cce67~
APPDIR=$WORKSPACE/object_detection_flutter $WORKSPACE/tflite_ios_bisect.sh
```

You can also run `tflite_ios_bisect.sh` with `git bisect` as documented at the top of the file. Theoretically you should land on the same breaking commit that I've indicated above...

**Other info / logs**

Relevant discussion at `tflite_flutter_plugin`: https://github.com/am15h/tflite_flutter_plugin/issues/64

**Comment by yyoon** [comment]
I've identified the same issue and fixed it in dff3c8a47a24b27f3e490f15f4de90743efc7ecf.
Can you sync your project after this commit and try again?

**Comment by mgalgs** [comment]
@yyoon that appears to have fixed it. Thank you!
